src/vcf/meta_entry.cpp

```diff
@@ -182,8 +182,42 @@ namespace ebi
             type_field != "String") {
             throw new MetaSectionError{entry.line, "FORMAT metadata Type is not a Integer, Float, Character or String"};
         }
+
+        if (entry.source->version == Version::v41 || entry.source->version == Version::v42) {
+            check_format_predefined_type(value, format_type_v41_v42);
+            check_format_predefined_number(value, format_number_v41_v42);
+        } else {
+            check_format_predefined_type(value, format_type_v43);
+            check_format_predefined_number(value, format_number_v43);
+        }
     }
-    
+
+    void MetaEntryVisitor::check_format_predefined_type(std::map<std::string, std::string> & value,
+                                                        std::map<std::string, std::pair<std::string, std::string>> const & format_types) const
+    {
+        auto & type_field = value["Type"];
+        for (const auto & type : format_types) {
```

Comment:
I just realised that we are iterating the whole map here, and we only want to check that type_field match if it is present in format_types. This also applies to the info methods. 

use the map::find to check if it is present, if it is not present, it is correct; if it present and doesn't match, throw error, if they match, it is correct